src/colmap/scene/reconstruction.cc

```diff
@@ -117,6 +117,13 @@ void Reconstruction::AddCamera(struct Camera camera) {
 }
 
 void Reconstruction::AddImage(class Image image) {
+  THROW_CHECK(image.HasCameraId());
+  const auto camera_it = cameras_.find(image.CameraId());
```

Comment:
Should we throw assertion error here on that the camera must exist

Reply:
Done.